### PR TITLE
Case sensitivity

### LINKED ISSUE #190: column operations do not maintain case sensitivity
### Describe the bug

From a user working with @lbenezra-FA 

> I'm working through contracts, and it looks like mesh is not recognizing case sensitivity. I've been looking through documentation, and I'm only seeing quoting as an option for database, schema, and table names. When the meshify tool automatically generates the contracts, is there a way to turn on case sensitivity? It depends on the project, but for some, we will have most of the fields needing it.

When we generate column entries in the contract step, we normalize to lowercase to match across the catalog and entries in the YML -- we should preserve case here!

### Expected results

Case is maintained!

### System information
**Which database are you using dbt with?**
- [ ] postgres
- [ ] redshift
- [ ] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)

**The output of `dbt debug`:**
```
<output goes here>
```

**The output of `dbt --version`:**
```
<output goes here>
```

### PR DESCRIPTION
**Pull request by dave-connors-3**
closes #190 

As reported, when creating a contract, all column names are currently standardized to lowercase, even if the column was in uppercase in the original yml, leading to unintended duplications/overwriting. This PR adds a step to the Contractor class to maintain the original casing of each of the columns from the yml entry. Anything not defined in yml, but present in the catalog will continue to be lowercased. 

I could not figure out a clean way to pass dynamic column content to our pytest fixture in the unit tests, so i created a new model fixture that included the column specifications. I think that's a big code smelly, so open to suggestions.

### CODE REVIEW

File: tests/unit/test_add_contract_to_yml.py [hunk 1]
```diff
@@ -54,12 +56,41 @@ def model() -> ModelNode:
...
 @pytest.fixture
 def change(project, model) -> ResourceChange:
     contractor = Contractor(project=project)
     return contractor.generate_contract(model)
 
 
+@pytest.fixture
+def all_cap_model_change(project, all_cap_model) -> ResourceChange:
+    contractor = Contractor(project=project)
+    return contractor.generate_contract(all_cap_model)
```
Comment (nicholasyager) [inline tests/unit/test_add_contract_to_yml.py:91]: This isn't terrible, but I see what you mean by code smell. I suspect that this fixture could follow a factory pattern instead, in which the fixture is a function that itself generates or returns models that have different capitalization configurations.

**Review by nicholasyager — CHANGES_REQUESTED** [review]
@dave-connors-3 I've left a few comments. One around cleaning up a print statement, and a couple of recommendations around nitpicky code-smell thoughts. Removing the print statement is the only blocking change in this review.

File: dbt_meshify/utilities/contractor.py [hunk 2]
```diff
@@ -15,11 +15,18 @@ def generate_contract(self, model: ModelNode) -> ResourceChange:
         """Generate a ChangeSet that adds a contract to a Model."""
         model_catalog = self.project.get_catalog_entry(model.unique_id)
 
+        # create a mapping of the column name to its representation in the yml file to maintain the original case
+        original_case = {column_name.lower(): column_name for column_name in model.columns.keys()}
+        print(original_case)
```
[the remark this answers is not in the document]
Comment (dave-connors-3) [reply dbt_meshify/utilities/contractor.py]: lol i caught this earlier and never pushed smdh

**Review by nicholasyager — APPROVED** [review]
@dave-connors-3 these changes are both necessary and sufficient. ✅

**Pull request merged**